sys/uri_parser/uri_parser.c

```diff
@@ -55,7 +55,7 @@ static char *_consume_scheme(uri_parser_result_t *result, char *uri,
     result->scheme_len = p - uri;
 
     /* check if authority part exists '://' */
-    if ((p[1] != '\0') && (p[2] != '\0') && (p[1] == '/') && (p[2] == '/')) {
+    if (((uri_end - p) > 3) && (p[1] == '/') && (p[2] == '/')) {
```

Comment:
Looks great (: do you mind adding a test for that similar to what was done here:
https://github.com/RIOT-OS/RIOT/pull/15929/files#diff-10c92785eb9572f526f95411c0fd4cf37fc6e39a62c8edda6261917db8253d3dR404
?

Reply:
Yes, will do tomorrow :)

Reply:
Added a test case, also changed `(uri_end - p) > 3` to `(uri_end - p) > 2`, otherwise the `://` in `pP://` is passed as a path. So thanks for suggestion the addition of a test case.